### DISCUSSION TITLE
upgradePassword method not called when using PasswordUpgraderInterface

### DISCUSSION THREAD

**Discussion by devabuse** [symfony/symfony#51846] [Q&A]
Application is running in PHP 8.1 and using Symfony 5.4.

I have the following configuration in my security.yaml:

    security:
       enable_authenticator_manager: true

       password_hashers:
          legacy:
             id: App\Account\Infrastructure\Framework\Symfony\Security\PasswordHasher

          App\Account\Domain\Model\Account:
            algorithm: argon2id
            migrate_from:
                - legacy

The custom password hasher implements a sha1 hashing algorithm with a salt (so the Account entity has a LegacyPasswordAuthenticatedUserInterface). It's not anything special.

The passwordHasher has the needsRehash return true and when going through the code with xDebug. I can clearly see it using a MigratingPasswordHandler. There's also a PasswordUpgrade badge in the badges which references the AccountRepository as implementing the PasswordUpgradeInterface.

I'm using a JWT endpoint which uses json_login and the Lexik JWT authentication bundle. I get a valid JWT token, so authentication has succeeded as well.

In the end though... It never calls the upgradePassword method in the repository.
 
At a loss here, what am I missing?

**Comment by MatTheCat** [comment]
Use Xdebug to step through `PasswordMigratingListener::onLoginSuccess`, this is the method which will migrate the password following a successful authentication.
